Move to protobufs

gkossakowski opened the pull request:
Changes explained in individual commits.

gkossakowski committed:
Move Jribble classes to separate package.

Moved jribble specific stuff to separate package.

Signed-off-by: Grzegorz Kossakowski <grzegorz.kossakowski@gmail.com>

gkossakowski committed:
Moved to protobuf-based jribble.

This is a large change but mostly straightforward one.

Specific changes are:

  * Nuke `scalagwt-lib` directory and references to it
    from build files and project files. We don't depend
    on any additional libs anymore!
  * JribbleParser delegates to protobuf test-based parsing
  * JribbleAstBuilder and JribbleReferenceMapper operate
    on protobuf messages now.
  * added jribble.proto file and added generation of
    JribbleProtos.java file by ant target called
    `generate.jribbleproto.java`. This is inspired
    by already existing ant target that generates
    java file for another proto file gwt includes.
  * updated tests. Note: AstUtils in tests has a lot of
    new helper methods for generating jribble asts. Jribble
    tests pass.

I'd like to mention that jribble tests turned out to be
very helpful when working on this move.
Thanks to @stephenh for contributing comprehensive set of tests.

gkossakowski committed:
Get rid of gwt.scalalibrary.path property.

We no longer have two differents versions of
Scala library on a classpath so we can get rid
of a hack that allowed us to load different
version of a library.

Signed-off-by: Grzegorz Kossakowski <grzegorz.kossakowski@gmail.com>

gkossakowski commented:
@stephenh, @lexspoon Can you have a look into this?

lexspoon commented:
Pull LGTM.

@stephenh, if you have time, a look at this patch would really help.

gkossakowski committed:
Clarified constructor-handling logic.

Per @lexpoon suggestion, clarified logic
handling constructors so it's easier to
find out what's going on and why we have
special cases here.

Signed-off-by: Grzegorz Kossakowski <grzegorz.kossakowski@gmail.com>
